virtualizarr/backend.py

```diff
@@ -133,9 +137,39 @@ def open_virtual_dataset(
         raise NotImplementedError()
 
     # if filetype is user defined, convert to FileType
+
     if filetype is not None:
         filetype = FileType(filetype)
 
+    if filetype == FileType.kerchunk_parquet:
+        from fsspec.implementations.reference import LazyReferenceMapper
+
+        from virtualizarr.readers.kerchunk import dataset_from_kerchunk_refs
+
+        fs = _FsspecFSFromFilepath(filepath=filepath, reader_options=reader_options).fs
+        lrm = LazyReferenceMapper(filepath, fs)
+
+        # build reference dict from KV pairs in LazyReferenceMapper
+        array_refs = {k: lrm[k] for k in lrm.keys()}
+
+        # where does version come from? Is it always 1?
+        full_reference = {"version": 1, "refs": array_refs}
+
+        return dataset_from_kerchunk_refs(KerchunkStoreRefs(full_reference))
+
+    if filetype == FileType.kerchunk_json:
+        import ast
+
+        from virtualizarr.readers.kerchunk import dataset_from_kerchunk_refs
+
+        fpath = _FsspecFSFromFilepath(
+            filepath=filepath, reader_options=reader_options
+        ).open_file()
+
+        refs = ast.literal_eval(fpath.read().decode("utf-8"))
```

Comment:
is there a specific reason for preferring `ast.literal_eval` over `json.load` / `json.loads` (or the `ujson` version of that)?

Reply:
I think the Kerchunk reference isn't valid JSON, so reading using `ast` allowed me to eval it as a dict, but very open to suggestions!

Reply:
as far as I remember the various `translate` methods of the `kerchunk` objects return python dictionaries, and the [usual pattern](https://fsspec.github.io/kerchunk/tutorial.html) is to explicitly dump those to json (not sure if anything changed since I last looked).

Given that `ast.literal_eval` is supposed to be 4-5 times slower than `json.load*` (the reference is python 3.3, so the exact number may have changed), I believe we should try to avoid using `ast.literal_eval`.

Reply:
Good to know about performance hit! Thanks for the feedback @keewis , I got `ujson` working and `ast` replaced.